src/modules/common/components/Headroom.js

```diff
@@ -0,0 +1,141 @@
+    this.lastScrollTop = 0 // the scrollTop when the user last changed his scrolling direction
+  }
+
+  componentDidMount () {
+    window.addEventListener('scroll', this.handleEvent)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('scroll', this.handleEvent)
+  }
+
+  /**
+   * Calculates the new transform value depending on the currentScrollTop position
+   * @param currentScrollTop
+   * @returns {number} the new transform value between -this.props.scrollHeight and 0
+   */
+  calcTransform (currentScrollTop) {
+    // Calculate new transform value while only regarding delta ScrollTop:
+    const scrolledTransform = this.lastTransform + (this.lastScrollTop - currentScrollTop)
+    // Transform should be at least:
+    // -this.props.scrollHeight, so we don't transform the header higher than we allow
+    // this.props.pinStart - currentScrollTop, so don't transform the header until we scrolled below this.props.pinStart
+    const maxTransform = Math.max(-this.props.scrollHeight, scrolledTransform, this.props.pinStart - currentScrollTop)
+    // Also we never want to translate the header down, so it should always be <= 0.
+    return Math.min(0, maxTransform)
+  }
+
+  /**
+   * Calculates the new stickyTop position for the ancestor based on the currentScrollTop and the corresponding
+   * transform value of the
+   * @param currentScrollTop
+   * @param transform
+   * @returns {number}
+   */
+  calcStickyTop (currentScrollTop, transform) {
+    // Optimize glitching behaviour when scrolling down from the top of the page
+    const shouldControlStickyAncestor = this.direction === DOWNWARDS
+      ? this.props.pinStart + this.props.scrollHeight < currentScrollTop
+      : this.props.pinStart < currentScrollTop
+    return shouldControlStickyAncestor
+      ? this.props.height + transform
+      : this.props.height - this.props.scrollHeight
+  }
+
+  /**
+   * Updates this.lastTranform and this.lastScrollTop if the direction has changed
+   * @param direction the new direction
+   */
+  setDirection (direction) {
+    if (this.direction !== direction) {
+      this.lastTransform = this.state.transform
+      this.lastScrollTop = this.lastKnownScrollTop
+      this.direction = direction
+    }
+  }
+
+  /**
+   * Calculates the new transform and stickyTop value and updates the state
+   */
+  update () {
+    const currentScrollTop = Headroom.getScrollTop()
+    this.setDirection(this.lastKnownScrollTop < currentScrollTop ? DOWNWARDS : UPWARDS)
+
+    const transform = this.calcTransform(currentScrollTop)
+    const stickyTop = this.calcStickyTop(currentScrollTop, transform)
+
+    this.setState({transform, stickyTop})
+
+    this.lastKnownScrollTop = currentScrollTop
+  }
+
+  handleEvent () {
+    if (!this.ticking) {
+      this.ticking = true
+      raf(() => { // Request animation frame to optimize performance for dom changes
```

Comment:
raf is only a devDependencies and no dependency. Not sure if we really need the polyfil though. It is only included currently because of a bug of jest.